src/storage/mvcc/reader.rs

```diff
@@ -271,6 +271,13 @@ impl<S: Snapshot> MvccReader<S> {
             if write.start_ts == start_ts {
                 return Ok(Some((commit_ts, write.write_type)));
             }
+
+            // If we reach a commit version whose type is not Rollback and start ts is
+            // larger than the given start ts, stop searching.
+            if write.write_type != WriteType::Rollback && write.start_ts > start_ts {
```

Comment:
Seems it doesn't matter if it is a rollback or not because the start ts is ascending anyway?

Reply:
@huachaohuang 
start_ts is not ascending. In test case `[40_35 PUT, 30_25 PUT, 20_20 Rollback, 10_1 PUT, 5_5 Rollback]`, when getting commit_ts for start_ts = 1, it will fail because of meeting start_ts = 5 first.